### PR TITLE
Backport 2.28: Fix undefined behavior in bignum: NULL+0 and -most-negative-sint

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Fix two kinds of typically harmless undefined behavior in bignum. Trivial backport of https://github.com/Mbed-TLS/mbedtls/pull/6609.

## Gatekeeper checklist

- [x] **changelog** provided
- [x] **backport** of https://github.com/Mbed-TLS/mbedtls/pull/6609
- [x] **tests** provided

### CODE REVIEW

File: include/mbedtls/bignum.h [hunk 1]
```diff
@@ -182,6 +182,20 @@
...
+ * \brief The type of machine digits in a bignum, called _limbs_.
+ *
+ * This is always an unsigned integer type with no padding bits. The size
+ * is platform-dependent.
+ */
+
+/** \typedef mbedtls_mpi_sint
+ * \brief The signed type corresponding to #mbedtls_mpi_uint.
+ *
+ * This is always an signed integer type with no padding bits. The size
```
Comment (tom-cosgrove-arm) [inline include/mbedtls/bignum.h:195]: Grr, missed this in the original. Up to you whether you fix :)

```suggestion
 * This is always a signed integer type with no padding bits. The size
```

Comment (gilles-peskine-arm) [reply include/mbedtls/bignum.h:195]: This hinders neither comprehension nor grepping so I'd prefer to merge if the CI is happy.

Comment (tom-cosgrove-arm) [reply include/mbedtls/bignum.h:195]: I gave an approval anyway...

**Comment by gilles-peskine-arm** [comment]
Marking `single-reviewer` since this is a trivial backport.

**Pull request merged**